### PR TITLE
Make maintainer documentation more direct

### PR DESCRIPTION
**Pull request by kevincox**
Additionally fixes the "list of names and emails" to be a list of maintainer expressions.

A follow-up from the discussion in https://github.com/NixOS/nixpkgs/pull/96666

### CODE REVIEW

File: doc/stdenv/meta.xml
Comment (jtojnar) [inline doc/stdenv/meta.xml:192]: Maybe change `add each new maintainer` to `add yourself`?

Comment (kevincox) [inline doc/stdenv/meta.xml:192]:
```suggestion
      A list of the maintainers of this Nix expression. Maintainers are defined in <link xlink:href="https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix"><filename>nixpkgs/maintainers/maintainer-list.nix</filename></link>. There is no restriction to becoming a maintainer, just add yourself to that list in a separate commit titled 'maintainers: add alice', and reference maintainers with <literal>maintainers = with lib.maintainers; [ alice bob ]</literal>.
```

Comment (kevincox) [reply doc/stdenv/meta.xml]: Sure.

**Review by jonringer — APPROVED** [review]
LGTM after squashing

**Comment by worldofpeace** [comment]
OMG thank u.

**Comment by kevincox** [comment]
Haha, I didn't realize the change was that dramatic 😄

**Comment by worldofpeace** [comment]
> Haha, I didn't realize the change was that dramatic smile

Editing docbook sources is always dramatic :rofl:

**Pull request merged**
